File: scala-server/app/com/bolour/eagerwords/common/domain/StopInfo.scala
/*
 * Copyright 2017-2020 Azad Bolour
 * Licensed under GNU Affero General Public License v3.0 -
 *    https://github.com/azadbolour/eagerwords/blob/master/LICENSE.md
 */
package com.bolour.eagerwords.common.domain

/**
  * Information about a completed game.
  *
  * TODO. Use sealed abstract case class for different reasons the game stopped.
  *
  * @param successivePasses Number of successive passes (was maxed out).
  * @param filledBoard The board has been filled and no more play is possible.
  */
case class StopInfo(
  successivePasses: Int,
  filledBoard: Boolean
)

File: haskell-server/src/EagerWords/Common/Domain/StopInfo.hs
--
-- Copyright 2017-2020 Azad Bolour
-- Licensed under GNU Affero General Public License v3.0 -
--   https://github.com/azadbolour/eagerwords/blob/master/LICENSE.md
--

{-# LANGUAGE DeriveGeneric #-}
{-# LANGUAGE DeriveAnyClass #-}

module EagerWords.Common.Domain.StopInfo (
  StopInfo(..)
) where

import GHC.Generics (Generic)
import Control.DeepSeq (NFData)
import Data.Aeson (FromJSON, ToJSON)

data StopInfo = StopInfo {
    successivePasses :: Int
  , filledBoard :: Bool
}
  deriving (Eq, Show, Generic, NFData)

instance FromJSON StopInfo
instance ToJSON StopInfo

